python/ray/tune/examples/tf_mnist_example.py

```diff
@@ -92,6 +92,12 @@ def test_step(images, labels):
         self.tf_train_step = train_step
         self.tf_test_step = test_step
 
+    def save_checkpoint(self, checkpoint_dir: str):
+        return None
+
+    def load_checkpoint(self, checkpoint):
+        return None
```

Comment:
I see -- is this the only place that needs to be updated?

Reply:
I'm actually surprised this is not happening in more cases (and why it didn't break before) - class trainables default to `checkpoint_at_end=True` and this can raise a `NotImplementedError`.

I think there are two options to solve this. 1) Default to `False` even for class trainables, or 2) Only set `True` if checkpointing is implemented (method is overwritten).

Reply:
Looked into this and turns out the default `checkpoint_at_end` is actually `None`.

Only in the `Tuner` do we set it to `True` for class trainables. In `tune.run` it just stays as `None` and no checkpoint happens at the end.

```python
from ray import tune


class Test(tune.Trainable):
    def step(self):
        return {"done": True}

tune.run(Test)  # works
tune.Tuner(Test).fit()  # errors
```